modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/RegistryLookupValueResolverWrapper.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.runtime.resolver;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+import static org.mule.runtime.api.util.Preconditions.checkArgument;
+import static org.mule.runtime.core.api.lifecycle.LifecycleUtils.initialiseIfNeeded;
+
+import org.mule.runtime.api.artifact.Registry;
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.lifecycle.Initialisable;
+import org.mule.runtime.api.lifecycle.InitialisationException;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.config.ConfigurationException;
+import org.mule.runtime.core.api.context.MuleContextAware;
+import org.mule.runtime.core.api.util.func.Once;
+
+import javax.inject.Inject;
+
+/**
+ * Implementation of {@link ValueResolver} which accesses the mule registry and returns the value associated with the result of
+ * another {@link ValueResolver}.
+ * <p/>
+ * Although the registry is mutable, the value associated to a given key is not meant to change. As a consequence,
+ * {@link #isDynamic()} will delegate to the wrapped {@link ValueResolver}.
+ *
+ * @since 4.5.0
+ */
+public class RegistryLookupValueResolverWrapper<T> implements ValueResolver<T>, Initialisable, MuleContextAware {
+
+  private final ValueResolver<String> delegate;
+  private MuleContext muleContext;
+  private Registry registry;
+
+  /**
+   * Construct a new instance and set the {@link ValueResolver} of the key that will be used to access the registry
+   *
+   * @param delegate a resolver for a not blank {@link String}
+   */
+  public RegistryLookupValueResolverWrapper(ValueResolver<String> delegate) {
```

Comment:
`delegate` -> `keysResolver`

Reply:
I removed this class in the latest version.